python/taichi/lang/matrix.py

```diff
@@ -522,6 +519,12 @@ def diag(dim, val):
     def loop_range(self):
         return self.entries[0]
 
+    def parent(self, n=1):
+        return self.loop_range().parent(n)
+
+    def snode(self):
+        return self.loop_range().snode()
+
```

Comment:
These two functions can be confusing: a Matrix can have many `SNode`s. I think exposing `loop_range()` is good enough.